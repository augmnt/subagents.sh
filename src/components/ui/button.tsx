import * as React from 'react';
import { Slot } from '@radix-ui/react-slot';
import { cva, type VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/utils';

const buttonVariants = cva(
  "inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono",
  {
    variants: {
      variant: {
        default:
          'bg-primary text-primary-foreground rounded-lg hover:bg-primary/90 hover:shadow-lg hover:-translate-y-0.5 active:translate-y-0 shadow-sm',
        terminal:
          'bg-terminal-green text-terminal-green-foreground rounded-lg hover:bg-terminal-green/90 hover:shadow-lg hover:-translate-y-0.5 active:translate-y-0 shadow-sm border-glow',
        destructive:
          'bg-destructive text-white rounded-lg hover:bg-destructive/90 hover:shadow-lg hover:-translate-y-0.5 active:translate-y-0 shadow-sm',
        outline:
          'border border-border bg-background rounded-lg hover:bg-accent hover:text-accent-foreground hover:shadow-md hover:-translate-y-0.5 active:translate-y-0 shadow-sm',
        secondary:
          'bg-secondary text-secondary-foreground rounded-lg hover:bg-secondary/80 hover:shadow-md hover:-translate-y-0.5 active:translate-y-0 shadow-sm',
        ghost:
          'rounded-lg hover:bg-accent hover:text-accent-foreground hover:shadow-sm',
        link: 'text-primary underline-offset-4 hover:underline',
      },
      size: {
        default: 'h-10 px-4 py-2 has-[>svg]:px-3',
        sm: 'h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5 text-xs',
        lg: 'h-12 rounded-lg px-6 has-[>svg]:px-4 text-base',
        icon: 'size-10',
      },
    },
    defaultVariants: {
      variant: 'default',
      size: 'default',
    },
  }
);

function Button({
  className,
  variant,
  size,
  asChild = false,
  ...props
}: React.ComponentProps<'button'> &
  VariantProps<typeof buttonVariants> & {
    asChild?: boolean;
  }) {
  const Comp = asChild ? Slot : 'button';

  return (
    <Comp
      data-slot="button"
      className={cn(buttonVariants({ variant, size, className }))}
      {...props}
    />
  );
}

export { Button, buttonVariants };
