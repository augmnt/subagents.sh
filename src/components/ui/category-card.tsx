'use client';

import * as React from 'react';
import { motion } from 'framer-motion';
import { 
  Terminal, 
  Code, 
  Database, 
  Globe, 
  Bot, 
  Zap, 
  Shield, 
  Search,
  ChevronRight,
  ArrowUpRight
} from 'lucide-react';
import { Badge } from './badge';
import { cn } from '@/lib/utils';

interface CategoryCardProps {
  name: string;
  description: string;
  count: number;
  icon?: string;
  featured?: boolean;
  className?: string;
  onClick?: () => void;
}

// Icon mapping for categories
const categoryIcons = {
  terminal: Terminal,
  code: Code,
  database: Database,
  web: Globe,
  ai: Bot,
  automation: Zap,
  security: Shield,
  search: Search,
} as const;

const CategoryCard = React.forwardRef<HTMLDivElement, CategoryCardProps>(
  ({
    name,
    description,
    count,
    icon = 'terminal',
    featured = false,
    className,
    onClick,
    ...props
  }, ref) => {
    const [isHovered, setIsHovered] = React.useState(false);
    const IconComponent = categoryIcons[icon as keyof typeof categoryIcons] || Terminal;

    return (
      <motion.div
        ref={ref}
        initial={{ opacity: 0, scale: 0.95 }}
        animate={{ opacity: 1, scale: 1 }}
        whileHover={{ 
          scale: 1.02,
          transition: { duration: 0.2, ease: "easeOut" }
        }}
        whileTap={{ scale: 0.98 }}
        className={cn(
          "group relative cursor-pointer overflow-hidden rounded-lg border bg-card p-6 shadow-sm transition-all duration-300",
          "hover:shadow-lg hover:border-terminal-green/50",
          featured && "border-terminal-green/30 bg-gradient-to-br from-card to-terminal-green/5",
          className
        )}
        onMouseEnter={() => setIsHovered(true)}
        onMouseLeave={() => setIsHovered(false)}
        onClick={onClick}
        {...props}
      >
        {/* Featured indicator */}
        {featured && (
          <div className="absolute top-3 right-3">
            <div className="w-2 h-2 bg-terminal-green rounded-full animate-pulse" />
          </div>
        )}

        {/* Icon */}
        <div className="mb-4 flex items-center justify-between">
          <motion.div
            initial={{ scale: 1, rotate: 0 }}
            animate={{ 
              scale: isHovered ? 1.1 : 1,
              rotate: isHovered ? 5 : 0
            }}
            transition={{ duration: 0.2 }}
            className={cn(
              "flex h-12 w-12 items-center justify-center rounded-lg border transition-all duration-300",
              "bg-gradient-to-br from-terminal-green/10 to-terminal-green/5",
              "border-terminal-green/20 group-hover:border-terminal-green/40",
              "group-hover:shadow-lg group-hover:shadow-terminal-green/20"
            )}
          >
            <IconComponent className="h-6 w-6 text-terminal-green" />
          </motion.div>

          <motion.div
            initial={{ opacity: 0, x: 20 }}
            animate={{ 
              opacity: isHovered ? 1 : 0,
              x: isHovered ? 0 : 20
            }}
            transition={{ duration: 0.2 }}
          >
            <ArrowUpRight className="h-5 w-5 text-muted-foreground" />
          </motion.div>
        </div>

        {/* Content */}
        <div className="space-y-2">
          <div className="flex items-center justify-between">
            <h3 className="font-mono text-lg font-semibold text-foreground group-hover:text-terminal-green transition-colors">
              {name}
            </h3>
            <Badge variant="ghost" className="text-xs">
              {count} agents
            </Badge>
          </div>
          
          <p className="text-sm text-muted-foreground line-clamp-2">
            {description}
          </p>
        </div>

        {/* Terminal-style bottom border */}
        <motion.div
          className="absolute bottom-0 left-0 h-0.5 bg-gradient-to-r from-terminal-green/0 via-terminal-green to-terminal-green/0"
          initial={{ width: "0%" }}
          animate={{ width: isHovered ? "100%" : "0%" }}
          transition={{ duration: 0.3, ease: "easeInOut" }}
        />

        {/* Hover overlay */}
        <motion.div
          className="absolute inset-0 bg-gradient-to-br from-terminal-green/5 to-transparent pointer-events-none"
          initial={{ opacity: 0 }}
          animate={{ opacity: isHovered ? 1 : 0 }}
          transition={{ duration: 0.3 }}
        />

        {/* Terminal-style dots pattern */}
        <div className="absolute top-3 left-3 flex gap-1 opacity-30">
          <div className="w-1 h-1 bg-terminal-green rounded-full" />
          <div className="w-1 h-1 bg-terminal-green rounded-full" />
          <div className="w-1 h-1 bg-terminal-green rounded-full" />
        </div>
      </motion.div>
    );
  }
);

CategoryCard.displayName = 'CategoryCard';

export { CategoryCard, type CategoryCardProps };