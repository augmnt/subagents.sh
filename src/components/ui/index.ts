// Core UI Components
export { Button, buttonVariants, type ButtonProps } from './button';
export { Badge, badgeVariants, type BadgeProps } from './badge';
export { Progress } from './progress';

// Terminal-Inspired Components  
export { AgentCard, type AgentCardProps } from './agent-card';
export { CategoryCard, type CategoryCardProps } from './category-card';
export { SearchBar, type SearchBarProps, type Filter } from './search-bar';
export { Rating, StarRatingInput, type RatingProps, type StarRatingInputProps } from './rating';

// Loading & Skeleton Components
export { 
  LoadingSpinner, 
  TerminalLoading, 
  LoadingOverlay,
  type LoadingSpinnerProps,
  type TerminalLoadingProps,
  type LoadingOverlayProps
} from './loading';

export { 
  Skeleton,
  AgentCardSkeleton,
  CategoryCardSkeleton,
  SearchBarSkeleton,
  ListSkeleton,
  TerminalSkeleton,
  type SkeletonProps,
  type ListSkeletonProps
} from './skeleton';

// Layout Components
export { 
  PageLayout, 
  PageTransition, 
  Section, 
  AnimatedGrid,
  type PageLayoutProps,
  type PageTransitionProps,
  type SectionProps,
  type AnimatedGridProps
} from '../layout/page-layout';