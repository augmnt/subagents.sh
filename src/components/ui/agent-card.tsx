'use client';

import * as React from 'react';
import Link from 'next/link';
import { Star, Eye, GitFork, Calendar, User, Tag, Github, ArrowRight, TrendingUp, Verified } from 'lucide-react';
import { Button } from './button';
import { Badge } from './badge';
import { cn } from '@/lib/utils';

interface AgentCardProps {
  id: string;
  title: string;
  description: string;
  author: string;
  authorUsername?: string;
  category: string;
  tags: string[];
  rating: number;
  downloads: number;
  views: number;
  forks: number;
  lastUpdated: string;
  codePreview?: string;
  verified?: boolean;
  featured?: boolean;
  isGitHubAuthor?: boolean;
  authorAvatar?: string;
  authorGitHubUrl?: string;
  className?: string;
  onClick?: () => void;
}

// Generate unique placeholder images for different components
const generatePlaceholderImage = (width: number, height: number, seed?: string) => {
  const baseUrl = 'https://picsum.photos';
  const seedParam = seed ? `?random=${seed}` : `?random=${Math.floor(Math.random() * 1000)}`;
  return `${baseUrl}/${width}/${height}${seedParam}`;
};

const AgentCard = React.forwardRef<HTMLDivElement, AgentCardProps>(
  ({
    id,
    title,
    description,
    author,
    authorUsername,
    category,
    tags,
    rating,
    downloads,
    views,
    forks,
    lastUpdated,
    codePreview,
    verified = false,
    featured = false,
    isGitHubAuthor = false,
    authorAvatar,
    authorGitHubUrl,
    className,
    onClick,
    ...props
  }, ref) => {
    const [showFullCode, setShowFullCode] = React.useState(false);


    // Format stats with clean display
    const formatStat = (value: number) => {
      if (value >= 1000000) return `${(value / 1000000).toFixed(1)}M`;
      if (value >= 1000) return `${(value / 1000).toFixed(1)}k`;
      return value.toLocaleString();
    };

    return (
      <div
        ref={ref}
        className={cn(
          "group relative cursor-pointer overflow-hidden rounded-xl transition-all duration-500 h-full flex flex-col",
          "bg-gradient-to-br from-card via-card/98 to-card/95 backdrop-blur-md",
          "border border-border/60 hover:border-border/80",
          "hover:shadow-2xl hover:shadow-primary/15 hover:scale-[1.03]",
          // Featured styling - subtle glow without border
          featured && [
            "shadow-lg shadow-amber-500/20",
            "bg-gradient-to-br from-amber-50/5 via-card/98 to-amber-50/5",
            "hover:shadow-amber-500/30",
          ],
          className
        )}
        onClick={onClick}
        {...props}
      >
        {/* Clean gradient overlay on hover */}
        <div className={cn(
          "absolute inset-0 bg-gradient-to-br from-primary/8 via-muted/10 to-purple-500/8 opacity-0 group-hover:opacity-100 transition-opacity duration-500",
          featured && "from-amber-400/10 via-amber-100/5 to-amber-500/10"
        )} />

        {/* Featured corner accent */}
        {featured && (
          <div className="absolute top-0 right-0 w-0 h-0 border-l-[20px] border-l-transparent border-t-[20px] border-t-amber-400/60" />
        )}

        {/* Card Content */}
        <div className="relative px-4 py-5 flex-1 flex flex-col">
          {/* Header Section */}
          <div className="mb-4">
            <div className="flex items-start justify-between mb-4">
              {/* Category Badge with Featured indicator */}
              <div className="flex items-center gap-2">
                <Badge className="bg-primary/15 text-primary border-0 font-semibold px-3 py-1">
                  {category}
                </Badge>
                {featured && (
                  <div className="flex items-center">
                    <Star className="h-3 w-3 text-amber-500 fill-amber-500" />
                    <span className="text-xs font-medium text-amber-600 ml-1">Featured</span>
                  </div>
                )}
              </div>

              {/* Rating */}
              <div className="flex items-center gap-1.5">
                <Star className="h-4 w-4 text-yellow-500 stroke-2" />
                <span className="text-sm font-semibold text-foreground">{rating.toFixed(1)}</span>
              </div>
            </div>
            
            <h3 className="text-xl font-bold group-hover:text-primary transition-all duration-300 leading-tight mb-3 text-foreground/90">
              {title}
            </h3>
            
            {/* Description */}
            <p className="text-muted-foreground/80 leading-relaxed text-sm mb-4">
              {description.length > 100 ? description.substring(0, 100) + '...' : description}
            </p>
          </div>

          {/* Tags - simplified */}
          <div className="flex flex-wrap gap-1.5 mb-6">
            {tags.slice(0, 3).map((tag) => (
              <Badge 
                key={tag} 
                variant="outline" 
                className="text-xs bg-muted/20 hover:bg-muted/40 border-border/30 transition-all duration-300 hover:scale-105"
              >
                {tag}
              </Badge>
            ))}
          </div>

          {/* Footer - simplified */}
          <div className="flex items-center justify-between mt-auto">
            <div className="flex items-center gap-4">
              {/* Author */}
              <div className="flex items-center gap-2">
                {authorUsername ? (
                  <Link 
                    href={`/user/${authorUsername}`}
                    className="flex items-center gap-2 hover:scale-105 transition-transform duration-200"
                    onClick={(e) => e.stopPropagation()}
                  >
                    <div className="w-6 h-6 flex-shrink-0">
                      {authorAvatar ? (
                        <img 
                          src={authorAvatar || generatePlaceholderImage(40, 40, author)}
                          alt={author}
                          className="w-full h-full rounded-full border border-border/30 object-cover cursor-pointer hover:border-primary/50"
                        />
                      ) : (
                        <div className="w-full h-full rounded-full bg-gradient-to-br from-primary/20 to-primary/5 flex items-center justify-center cursor-pointer hover:bg-gradient-to-br hover:from-primary/30 hover:to-primary/15">
                          <User className="w-3 h-3 text-primary" />
                        </div>
                      )}
                    </div>
                    <span className="text-xs font-medium text-foreground/70 truncate max-w-[80px] sm:max-w-[120px] cursor-pointer hover:text-primary transition-colors duration-200">
                      {author}
                    </span>
                  </Link>
                ) : (
                  <>
                    <div className="w-6 h-6 flex-shrink-0">
                      {authorAvatar ? (
                        <img 
                          src={authorAvatar || generatePlaceholderImage(40, 40, author)}
                          alt={author}
                          className="w-full h-full rounded-full border border-border/30 object-cover"
                        />
                      ) : (
                        <div className="w-full h-full rounded-full bg-gradient-to-br from-primary/20 to-primary/5 flex items-center justify-center">
                          <User className="w-3 h-3 text-primary" />
                        </div>
                      )}
                    </div>
                    <span className="text-xs font-medium text-foreground/70 truncate max-w-[80px] sm:max-w-[120px]">
                      {author}
                    </span>
                  </>
                )}
              </div>
              
              {/* Downloads */}
              <div className="flex items-center gap-1">
                <TrendingUp className="h-3.5 w-3.5 text-green-600" />
                <span className="text-xs font-medium text-muted-foreground">
                  {formatStat(downloads)}
                </span>
              </div>
            </div>
            
            <Button 
              size="sm" 
              className="bg-gradient-to-r from-primary to-primary/80 hover:from-primary/90 hover:to-primary/70 text-primary-foreground shadow-lg hover:shadow-xl transition-all duration-300 group-hover:scale-105"
            >
              <span className="font-medium">View</span>
              <ArrowRight className="ml-1.5 h-3.5 w-3.5 group-hover:translate-x-0.5 transition-transform duration-300" />
            </Button>
          </div>
        </div>
      </div>
    );
  }
);

AgentCard.displayName = 'AgentCard';

export { AgentCard, type AgentCardProps };