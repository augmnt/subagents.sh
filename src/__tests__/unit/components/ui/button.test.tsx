import React from 'react';
import { render, screen } from '@/__tests__/utils/test-utils';
import userEvent from '@testing-library/user-event';
import { Button } from '@/components/ui/button';
import { checkAccessibility } from '@/__tests__/utils/test-utils';

describe('Button Component', () => {
  it('renders with default props', () => {
    render(<Button>Click me</Button>);
    
    const button = screen.getByRole('button', { name: /click me/i });
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('inline-flex', 'items-center', 'justify-center');
  });

  it('renders with different variants', () => {
    const { rerender } = render(<Button variant="default">Default</Button>);
    expect(screen.getByRole('button')).toHaveClass('bg-primary');

    rerender(<Button variant="terminal">Terminal</Button>);
    expect(screen.getByRole('button')).toHaveClass('bg-terminal-green');

    rerender(<Button variant="destructive">Destructive</Button>);
    expect(screen.getByRole('button')).toHaveClass('bg-destructive');

    rerender(<Button variant="outline">Outline</Button>);
    expect(screen.getByRole('button')).toHaveClass('border', 'border-border');

    rerender(<Button variant="secondary">Secondary</Button>);
    expect(screen.getByRole('button')).toHaveClass('bg-secondary');

    rerender(<Button variant="ghost">Ghost</Button>);
    expect(screen.getByRole('button')).not.toHaveClass('bg-primary');

    rerender(<Button variant="link">Link</Button>);
    expect(screen.getByRole('button')).toHaveClass('text-primary', 'underline-offset-4');
  });

  it('renders with different sizes', () => {
    const { rerender } = render(<Button size="default">Default</Button>);
    expect(screen.getByRole('button')).toHaveClass('h-10', 'px-4');

    rerender(<Button size="sm">Small</Button>);
    expect(screen.getByRole('button')).toHaveClass('h-8', 'px-3', 'text-xs');

    rerender(<Button size="lg">Large</Button>);
    expect(screen.getByRole('button')).toHaveClass('h-12', 'px-6', 'text-base');

    rerender(<Button size="icon">Icon</Button>);
    expect(screen.getByRole('button')).toHaveClass('size-10');
  });

  it('handles click events', async () => {
    const handleClick = jest.fn();
    const user = userEvent.setup();
    
    render(<Button onClick={handleClick}>Click me</Button>);
    
    const button = screen.getByRole('button', { name: /click me/i });
    await user.click(button);
    
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('can be disabled', async () => {
    const handleClick = jest.fn();
    const user = userEvent.setup();
    
    render(<Button onClick={handleClick} disabled>Disabled Button</Button>);
    
    const button = screen.getByRole('button', { name: /disabled button/i });
    expect(button).toBeDisabled();
    expect(button).toHaveClass('disabled:pointer-events-none', 'disabled:opacity-50');
    
    await user.click(button);
    expect(handleClick).not.toHaveBeenCalled();
  });

  it('supports custom className', () => {
    render(<Button className="custom-class">Custom</Button>);
    
    const button = screen.getByRole('button', { name: /custom/i });
    expect(button).toHaveClass('custom-class');
  });

  it('renders as child component when asChild is true', () => {
    render(
      <Button asChild>
        <a href="/test">Link Button</a>
      </Button>
    );
    
    const link = screen.getByRole('link', { name: /link button/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', '/test');
    expect(link).toHaveClass('inline-flex', 'items-center');
  });

  it('forwards ref correctly', () => {
    const ref = React.createRef<HTMLButtonElement>();
    
    render(<Button ref={ref}>Button with ref</Button>);
    
    expect(ref.current).toBeInstanceOf(HTMLButtonElement);
    expect(ref.current).toHaveTextContent('Button with ref');
  });

  it('supports aria attributes', () => {
    render(
      <Button aria-label="Custom label" aria-describedby="description">
        Button
      </Button>
    );
    
    const button = screen.getByRole('button', { name: /custom label/i });
    expect(button).toHaveAttribute('aria-describedby', 'description');
  });

  it('has proper focus styles', async () => {
    const user = userEvent.setup();
    
    render(<Button>Focus me</Button>);
    
    const button = screen.getByRole('button', { name: /focus me/i });
    await user.tab();
    
    expect(button).toHaveFocus();
    expect(button).toHaveClass('focus-visible:ring-2');
  });

  it('passes accessibility checks', async () => {
    const { container } = render(<Button>Accessible Button</Button>);
    await checkAccessibility(container);
  });

  it('handles keyboard navigation', async () => {
    const handleClick = jest.fn();
    const user = userEvent.setup();
    
    render(<Button onClick={handleClick}>Keyboard Button</Button>);
    
    const button = screen.getByRole('button', { name: /keyboard button/i });
    await user.tab();
    expect(button).toHaveFocus();
    
    await user.keyboard('{Enter}');
    expect(handleClick).toHaveBeenCalledTimes(1);
    
    await user.keyboard('{Space}');
    expect(handleClick).toHaveBeenCalledTimes(2);
  });

  it('applies data-slot attribute', () => {
    render(<Button>Data Slot</Button>);
    
    const button = screen.getByRole('button', { name: /data slot/i });
    expect(button).toHaveAttribute('data-slot', 'button');
  });

  describe('Button Variants Visual Regression', () => {
    it('renders all variants consistently', () => {
      const variants = ['default', 'terminal', 'destructive', 'outline', 'secondary', 'ghost', 'link'] as const;
      
      variants.forEach(variant => {
        const { unmount } = render(<Button variant={variant}>{variant} Button</Button>);
        const button = screen.getByRole('button');
        expect(button).toMatchSnapshot(`button-variant-${variant}`);
        unmount();
      });
    });

    it('renders all sizes consistently', () => {
      const sizes = ['default', 'sm', 'lg', 'icon'] as const;
      
      sizes.forEach(size => {
        const { unmount } = render(<Button size={size}>{size} Button</Button>);
        const button = screen.getByRole('button');
        expect(button).toMatchSnapshot(`button-size-${size}`);
        unmount();
      });
    });
  });

  describe('Button Edge Cases', () => {
    it('handles empty children', () => {
      render(<Button></Button>);
      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();
    });

    it('handles multiple children', () => {
      render(
        <Button>
          <span>Icon</span>
          <span>Text</span>
        </Button>
      );
      
      const button = screen.getByRole('button');
      expect(button).toHaveTextContent('IconText');
    });

    it('preserves other button attributes', () => {
      render(
        <Button 
          type="submit" 
          form="test-form" 
          data-testid="test-button"
          aria-pressed="true"
        >
          Submit
        </Button>
      );
      
      const button = screen.getByRole('button', { name: /submit/i });
      expect(button).toHaveAttribute('type', 'submit');
      expect(button).toHaveAttribute('form', 'test-form');
      expect(button).toHaveAttribute('data-testid', 'test-button');
      expect(button).toHaveAttribute('aria-pressed', 'true');
    });
  });
});