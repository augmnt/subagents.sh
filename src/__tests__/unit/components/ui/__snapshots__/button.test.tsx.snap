// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button Variants Visual Regression renders all sizes consistently: button-size-default 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono bg-primary text-primary-foreground rounded-lg hover:bg-primary/90 hover:shadow-lg hover:-translate-y-0.5 active:translate-y-0 shadow-sm h-10 px-4 py-2 has-[>svg]:px-3"
  data-slot="button"
>
  default
   Button
</button>
`;

exports[`Button Component Button Variants Visual Regression renders all sizes consistently: button-size-icon 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono bg-primary text-primary-foreground rounded-lg hover:bg-primary/90 hover:shadow-lg hover:-translate-y-0.5 active:translate-y-0 shadow-sm size-10"
  data-slot="button"
>
  icon
   Button
</button>
`;

exports[`Button Component Button Variants Visual Regression renders all sizes consistently: button-size-lg 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono bg-primary text-primary-foreground hover:bg-primary/90 hover:shadow-lg hover:-translate-y-0.5 active:translate-y-0 shadow-sm h-12 rounded-lg px-6 has-[>svg]:px-4 text-base"
  data-slot="button"
>
  lg
   Button
</button>
`;

exports[`Button Component Button Variants Visual Regression renders all sizes consistently: button-size-sm 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono bg-primary text-primary-foreground hover:bg-primary/90 hover:shadow-lg hover:-translate-y-0.5 active:translate-y-0 shadow-sm h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5 text-xs"
  data-slot="button"
>
  sm
   Button
</button>
`;

exports[`Button Component Button Variants Visual Regression renders all variants consistently: button-variant-default 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono bg-primary text-primary-foreground rounded-lg hover:bg-primary/90 hover:shadow-lg hover:-translate-y-0.5 active:translate-y-0 shadow-sm h-10 px-4 py-2 has-[>svg]:px-3"
  data-slot="button"
>
  default
   Button
</button>
`;

exports[`Button Component Button Variants Visual Regression renders all variants consistently: button-variant-destructive 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono bg-destructive text-white rounded-lg hover:bg-destructive/90 hover:shadow-lg hover:-translate-y-0.5 active:translate-y-0 shadow-sm h-10 px-4 py-2 has-[>svg]:px-3"
  data-slot="button"
>
  destructive
   Button
</button>
`;

exports[`Button Component Button Variants Visual Regression renders all variants consistently: button-variant-ghost 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono rounded-lg hover:bg-accent hover:text-accent-foreground hover:shadow-sm h-10 px-4 py-2 has-[>svg]:px-3"
  data-slot="button"
>
  ghost
   Button
</button>
`;

exports[`Button Component Button Variants Visual Regression renders all variants consistently: button-variant-link 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono text-primary underline-offset-4 hover:underline h-10 px-4 py-2 has-[>svg]:px-3"
  data-slot="button"
>
  link
   Button
</button>
`;

exports[`Button Component Button Variants Visual Regression renders all variants consistently: button-variant-outline 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono border border-border bg-background rounded-lg hover:bg-accent hover:text-accent-foreground hover:shadow-md hover:-translate-y-0.5 active:translate-y-0 shadow-sm h-10 px-4 py-2 has-[>svg]:px-3"
  data-slot="button"
>
  outline
   Button
</button>
`;

exports[`Button Component Button Variants Visual Regression renders all variants consistently: button-variant-secondary 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono bg-secondary text-secondary-foreground rounded-lg hover:bg-secondary/80 hover:shadow-md hover:-translate-y-0.5 active:translate-y-0 shadow-sm h-10 px-4 py-2 has-[>svg]:px-3"
  data-slot="button"
>
  secondary
   Button
</button>
`;

exports[`Button Component Button Variants Visual Regression renders all variants consistently: button-variant-terminal 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-2 focus-visible:ring-terminal-green/50 focus-visible:ring-offset-2 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-mono bg-terminal-green text-terminal-green-foreground rounded-lg hover:bg-terminal-green/90 hover:shadow-lg hover:-translate-y-0.5 active:translate-y-0 shadow-sm border-glow h-10 px-4 py-2 has-[>svg]:px-3"
  data-slot="button"
>
  terminal
   Button
</button>
`;
